shell/common/rasterizer.cc

```diff
@@ -270,13 +271,23 @@ sk_sp<SkImage> Rasterizer::DoMakeRasterSnapshot(
                 return;
               }
 
+              GrDirectContext* context = surface_->GetContext();
+
+              // Upper-bound to the max render target size supported by the GPU.
+              // Prefer to proceed with a smaller snapshot over failing with a
+              // null result.
+              auto maxSize = context->maxRenderTargetSize();
+              image_info =
+                  image_info.makeWH(std::min(maxSize, image_info.width()),
```

Comment:
This will not preserve the aspect ratio and clip contents. Scaling needs to be aspect ratio aware.

Reply:
Ah, of course. Thanks for catching that. I changed it to scale down properly.